tasks/version_powershell.ps1

```diff
@@ -1,14 +1,16 @@
-if (Test-Path 'HKLM:\SOFTWARE\Puppet Labs\Puppet') {
-  $reg = $(Get-ItemProperty -Path 'HKLM:\SOFTWARE\Puppet Labs\Puppet')
-  if (Test-Path $reg.RememberedInstallDir64) {
+$rootPath = 'HKLM:\SOFTWARE\Puppet Labs\Puppet'
+
+$reg = Get-ItemProperty -Path $rootPath -ErrorAction SilentlyContinue
+if ($null -ne $reg) {
+  if ($null -ne $reg.RememberedInstallDir64) {
     $loc = $reg.RememberedInstallDir64+'VERSION'
-  } elseif (Test-Path $reg.RememberedInstallDir) {
+  } elseif ($null -ne $reg.RememberedInstallDir) {
     $loc = $reg.RememberedInstallDir+'VERSION'
   }
 }
 
-if ($loc -ne $null) {
-  Write-Output "{`"version`":`"$(type $loc)`",`"source`":`"$($loc.replace('\', '/'))`"}"
+if ($null -ne $loc) {
+  Write-Output "{`"version`":`"$(Get-Content -Path $loc -ErrorAction SilentlyContinue)`",`"source`":`"$($loc.replace('\', '/'))`"}"
```

Comment:
`-ErrorAction SilentlyContinue` will result in an empty version if reading the file fails for some reason (permissions?). I'm not sure that's what we want... if the file is present but we can't read it, we may want to fail the task.

Reply:
Fair enough.  Can just read it first before `write-Output`